rotkehlchen/db/filtering.py

```diff
@@ -310,8 +322,17 @@ def make(
         if tx_hash is not None:  # tx_hash means single result so make it as single filter
             filters.append(DBETHTransactionHashFilter(and_op=False, tx_hash=tx_hash))
         else:
-            if addresses is not None:
-                filter_query.join_clause = DBETHTransactionAddressFilter(and_op=False, addresses=addresses)  # noqa: E501
+            if addresses or asset or protocols:
+                filter_query.join_clause = DBETHTransactionJoinsFilter(
+                    and_op=False,
+                    addresses=addresses,
+                    join_events=bool(asset or protocols),
```

Comment:
you already have this check above. So why not keep it in a variable and use it here too without redoing it?

Reply:
Yep, good idea